examples/deploy-contract.js

```diff
@@ -0,0 +1,88 @@
+
+/* Example contract:
+* // SPDX-License-Identifier: MIT
+* pragma solidity ^0.8.0;
+* 
+* contract SimpleStorage {
+*     uint256 private storedValue;
+* 
+*     event ValueSet(uint256 value, address indexed sender);
+* 
+*     function set(uint256 value) public {
+*         storedValue = value;
+*         emit ValueSet(value, msg.sender);
+*     }
+* 
+*     function get() public view returns (uint256) {
+*         uint256 value = storedValue;
+*         return value;
+*     }
+* }
+*
+*/
+
+const quais = require('../lib/commonjs/quais');
+const fs = require('fs');
+require('dotenv').config();
+
+// Prerequisites: compile the smart contract and generate artifacts, i.e. run:
+// solc --bin --abi SimpleStorage.sol --output-dir artifacts 
+
+const abi = JSON.parse(fs.readFileSync('artifacts/SimpleStorage.abi', 'utf8'));
+const bytecode = fs.readFileSync('artifacts/SimpleStorage.bin', 'utf8');
+
+async function main() {
+	const provider = new quais.JsonRpcProvider(process.env.RPC_URL);
+	
+	const wallet = new quais.Wallet(process.env.CYPRUS1PK, provider);
+	console.log('Wallet address:', wallet.address);
+
+	// get address balance
+	let balance = await provider.getBalance(wallet.address);
+	console.log('Balance:', balance.toString());
+	
+	// Deploy contract	
+	const factory = new quais.ContractFactory(abi, bytecode, wallet);
+	
+	let nonce = await provider.getTransactionCount(wallet.address, 'latest');
+	const deployParams = {
+		nonce,
+		gasLimit: 420000,
+		maxPriorityFeePerGas: 1000000000n,
+		maxFeePerGas: 3000000000000n,
+		from: wallet.address,
+	};
+	let contract = await factory.deploy(deployParams);
+	const contractAddress = await contract.getAddress();
+	console.log('Contract address:', contractAddress);
+	
+	// wait for contract to be deployed
+	const receipt = await contract.waitForDeployment(); 
+	console.log('receipt:', receipt);
```

Comment:
This console.log `receipt` terminology is misleading becasue `contract.waitForDeployment()` returns and instance of `BaseContract` rather than any sort of receipt